src/lib/frontend/typechecker.ml

```diff
@@ -2154,6 +2156,51 @@ let type_user_defined_type_body ~is_recursive env acc (loc, ls, s, body) =
     assert (not is_recursive); (* Abstract types are not recursive *)
     acc, env
 
+let type_fun_aux_1 env loc n ?ret_ty l  =
```

Comment:
is it possible to find a better name for this function (and for `type_fun_aux_2`) ? something that would better describe what the function is doing...
Something along the lines of `declare` for `aux_1` and define for `aux_2` if I understand the code correctly.

Reply:
Done.